### ISSUE TITLE
Mixer has random errors on restart in Istio 1.0.2

### ISSUE THREAD

**Issue by theganyo** [istio/istio#8839]
The Mixer in Istio 1.0.2 seems to have a race condition where it may or may not get various kinds errors during startup. This did not happen in Istio 1.0.0.

Reproducing is simple: 
1. Start Istio.
2. Delete the policy pod.
3. Check the policy pod's logs.

The logs will likely have errors during startup, but not necessarily and the errors can vary each time the pod is restarted.

**Version**
Istio Version: 1.0.2
Kubernetes version: 1.10.3

**Installation**
kubectl apply -f install/kubernetes/istio-demo.yaml

**Environment**
Docker for Mac 18.06.1-ce-mac74
OS X 10.13.6

Here's an example policy mixer log:

```
➜  kubectl -n istio-system delete po istio-policy-67f4d49564-tnwrv
pod "istio-policy-67f4d49564-tnwrv" deleted
➜  kubectl -n istio-system logs istio-policy-67f4d49564-ghjh9 mixer
Mixer started with
MaxMessageSize: 1048576
MaxConcurrentStreams: 1024
APIWorkerPoolSize: 1024
AdapterWorkerPoolSize: 1024
APIPort: 9091
APIAddress: unix:///sock/mixer.socket
MonitoringPort: 9093
EnableProfiling: true
SingleThreaded: false
NumCheckCacheEntries: 1500000
ConfigStoreURL: k8s://
ConfigDefaultNamespace: istio-system
LoggingOptions: log.Options{OutputPaths:[]string{"stdout"}, ErrorOutputPaths:[]string{"stderr"}, RotateOutputPath:"", RotationMaxSize:104857600, RotationMaxAge:30, RotationMaxBackups:1000, JSONEncoding:false, LogGrpc:true, outputLevels:"default:info", logCallers:"", stackTraceLevels:"default:none"}
TracingOptions: tracing.Options{ZipkinURL:"http://zipkin:9411/api/v1/spans", JaegerURL:"", LogTraceSpans:false}
IntrospectionOptions: ctrlz.Options{Port:0x2694, Address:"127.0.0.1"}

2018-09-19T17:26:22.692146Z	warn	Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
2018-09-19T17:26:22.692902Z	info	Built new config.Snapshot: id='0'
2018-09-19T17:26:22.693158Z	info	Cleaning up handler table, with config ID:-1
2018-09-19T17:26:22.713594Z	warn	Failed to discover kinds: [dogstatsd cloudwatch]
2018-09-19T17:26:22.714557Z	error	error evaluating AttributeBinding expression for attribute key 'destination.ip': unknown attribute destination.ip
2018-09-19T17:26:22.714592Z	info	Built new config.Snapshot: id='1'
2018-09-19T17:26:22.714796Z	info	Cleaning up handler table, with config ID:0
2018-09-19T17:26:22.714825Z	info	Awaiting for config store sync...
2018-09-19T17:26:23.908482Z	info	Publishing 9 events
2018-09-19T17:26:23.909062Z	error	failed to evaluate expression for field 'Variables[responseCode]'; unknown attribute response.code
2018-09-19T17:26:23.909232Z	error	failed to evaluate expression for field 'Variables[receivedBytes]'; unknown attribute connection.received.bytes
2018-09-19T17:26:23.909497Z	error	error evaluating AttributeBinding expression for attribute key 'destination.serviceAccount': unknown attribute destination.serviceAccount
2018-09-19T17:26:23.909696Z	error	unknown attribute context.protocol
2018-09-19T17:26:23.910050Z	error	Handler not found: handler='handler.stdio'
2018-09-19T17:26:23.910250Z	error	No valid actions found in rule
2018-09-19T17:26:23.910312Z	error	unknown attribute context.protocol
2018-09-19T17:26:23.910434Z	error	Instance not found: instance='requestcount.metric'
2018-09-19T17:26:23.910633Z	error	Instance not found: instance='requestduration.metric'
2018-09-19T17:26:23.910774Z	error	Instance not found: instance='requestsize.metric'
2018-09-19T17:26:23.910887Z	error	Instance not found: instance='responsesize.metric'
2018-09-19T17:26:23.910899Z	error	No valid instances found
2018-09-19T17:26:23.910995Z	error	No valid actions found in rule
2018-09-19T17:26:23.911133Z	error	Instance not found: instance='attributes.kubernetes'
2018-09-19T17:26:23.911266Z	error	No valid instances found
2018-09-19T17:26:23.911458Z	error	No valid actions found in rule
2018-09-19T17:26:23.911617Z	error	unknown attribute context.protocol
2018-09-19T17:26:23.911763Z	error	Instance not found: instance='attributes.kubernetes'
2018-09-19T17:26:23.911882Z	error	No valid instances found
2018-09-19T17:26:23.911969Z	error	No valid actions found in rule
2018-09-19T17:26:23.912004Z	error	unknown attribute context.protocol
2018-09-19T17:26:23.912020Z	error	Instance not found: instance='tcpbytesent.metric'
2018-09-19T17:26:23.912124Z	error	Instance not found: instance='tcpbytereceived.metric'
2018-09-19T17:26:23.912134Z	error	No valid instances found
2018-09-19T17:26:23.912157Z	error	No valid actions found in rule
2018-09-19T17:26:23.912252Z	error	unknown attribute context.protocol
2018-09-19T17:26:23.912264Z	error	Handler not found: handler='handler.stdio'
2018-09-19T17:26:23.912274Z	error	No valid actions found in rule
2018-09-19T17:26:23.912281Z	info	Built new config.Snapshot: id='2'
2018-09-19T17:26:23.912634Z	info	Cleaning up handler table, with config ID:1
2018-09-19T17:26:24.917122Z	info	Publishing 3 events
2018-09-19T17:26:24.918173Z	error	Instance not found: instance='requestcount.metric'
2018-09-19T17:26:24.918230Z	error	Instance not found: instance='requestduration.metric'
2018-09-19T17:26:24.918245Z	error	Instance not found: instance='requestsize.metric'
2018-09-19T17:26:24.918291Z	error	Instance not found: instance='responsesize.metric'
2018-09-19T17:26:24.918633Z	error	No valid instances found
2018-09-19T17:26:24.918652Z	error	No valid actions found in rule
2018-09-19T17:26:24.918932Z	error	Instance not found: instance='tcpbytesent.metric'
2018-09-19T17:26:24.918975Z	error	Instance not found: instance='tcpbytereceived.metric'
2018-09-19T17:26:24.918991Z	error	No valid instances found
2018-09-19T17:26:24.919005Z	error	No valid actions found in rule
2018-09-19T17:26:24.919111Z	info	Built new config.Snapshot: id='3'
2018-09-19T17:26:24.919587Z	info	adapters	getting kubeconfig from: ""	{"adapter": "handler.kubernetesenv.istio-system"}
2018-09-19T17:26:24.919628Z	warn	Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
2018-09-19T17:26:24.920799Z	info	adapters	Waiting for kubernetes cache sync...	{"adapter": "handler.kubernetesenv.istio-system"}
2018-09-19T17:26:25.021833Z	info	adapters	Cache sync successful.	{"adapter": "handler.kubernetesenv.istio-system"}
2018-09-19T17:26:25.023662Z	info	Cleaning up handler table, with config ID:2
2018-09-19T17:26:26.313144Z	info	Publishing 6 events
2018-09-19T17:26:26.318362Z	info	Built new config.Snapshot: id='4'
2018-09-19T17:26:26.323701Z	info	adapters	serving prometheus metrics on 42422	{"adapter": "handler.prometheus.istio-system"}
2018-09-19T17:26:26.324436Z	info	Cleaning up handler table, with config ID:3
```

**Comment by mandarjog** [comment]
@theganyo I am seeing similar error messages at start, however all configuration is loaded by the time `starting monitor server...` message comes about.
Are you seeing that istio-policy pod is broken and stays broken?

the cause is that we no longer wait 30s to start istio-policy. We process configuration as it comes in, there are some easy ways to reduce this noise.

**Comment by sdake** [comment]
@mandarjog in 1.0.2 and even in master, this causes telemetry not to report telemetry.  The only solution at this point is to kill the telemetry pod and let the deployment restart it (or killall -15 mixs).  The documented solution for this is to apply the crds ahead of applying the manifest (see: https://istio.io/docs/setup/kubernetes/quick-start/#installation-steps or https://istio.io/docs/setup/kubernetes/helm-install/#installation-steps).  IMO this is a fundamental bug in Kubernetes related to how CRDs are processed when a manifest is processed.  CRDs should be processed first and committed prior to processing other resources in the manifest.

 A simple solution for master would be to simply exit when the CRDs are not available.  Not sure if that would be a suitable workaround for this Kubernetes defect for the 1.0 series.

cc @douglas-reid

**Comment by douglas-reid** [comment]
@mandarjog Perhaps we should consider deploying Mixer with a fallback attribute set (for Envoy) baked in? Most deployments probably aren't adding / altering that set, and it would really go a long way to avoid timing issues and `unknown attribute destination.ip` related expression eval errors.

**Comment by mandarjog** [comment]
Exiting when built in crds are not found will fix the issue, it was on the roadmap, let’s do it now.
This takes care of the case when all crds and crs are actually available, but mixer started too soon.
The race between crds and cr is still an orthogonal issue, and mixer restart does not address that.

**Comment by theganyo** [comment]
"Are you seeing that istio-policy pod is broken and stays broken?" 
Uncertain. I've had user reports of 1.0.2 not working and falling back to 1.0.0 fixing the issue. After debugging, this is where I ended up. So Mixer reports errors and never explicitly reports that all errors are resolved and Mixer is in a good and healthy state. Thus, it leaves me concerned because I don't know the full extent of what might be working or what might not be working after receiving these errors. And, I don't know of an easy way to determine if Mixer is completely trustworthy at any point in time.

**Comment by mandarjog** [comment]
@bianpengyuan can you take a look at this?
At a certain point, we used to wait upto 30s to start Mixer. This was ample time for Mixer get all its configuration and start. Now we start mixer sooner.
1. Adding an info log to say that everything was loaded would also make sense.  
2. sdake reports that mixer stays broken, which is completely unexpected and it surely a bug. When the all the CRS are available, Mixer should eventually start working. @sdake Do you have logs from when Mixer did not start properly and stayed broken?

**Comment by theganyo** [comment]
I can corroborate that Mixer stays broken.

**Comment by hklai** [comment]
@bianpengyuan what is the status? If this still needed for 1.0.3?

**Comment by bianpengyuan** [comment]
The fix has been merged into 1.0 branch, and it is needed for 1.0.3.

**Comment by mandarjog** [comment]
@bianpengyuan
reports that the problem still persists.

 ```2018-10-01T16:24:44.776059Z	info Awaiting for config store sync...
gc 2 @0.052s 14%: 0.082+6.2+6.2 ms clock, 0.24+0.16/0.46/10+18 ms cpu, 4->4->3 MB, 5 MB goal, 3 P
2018-10-01T16:24:45.793115Z	info	Publishing 10 events
2018-10-01T16:24:45.794861Z	error	failed to evaluate expression for field 'Dimensions[source_version]'; unknown attribute source.labels
2018-10-01T16:24:45.795000Z	error	failed to evaluate expression for field 'Dimensions[source_version]'; unknown attribute source.labels
2018-10-01T16:24:45.795134Z	error	failed to evaluate expression for field 'Dimensions[destination_service_name]'; unknown attribute destination.service.name
2018-10-01T16:24:45.795295Z	error	failed to evaluate expression for field 'Dimensions[source_workload]'; unknown attribute source.workload.name
2018-10-01T16:24:45.795402Z	error	failed to evaluate expression for field 'Variables[sourceName]'; unknown attribute source.name
2018-10-01T16:24:45.795615Z	error	failed to evaluate expression for field 'Variables[sourceName]'; unknown attribute source.name
2018-10-01T16:24:45.795765Z	error	failed to evaluate expression for field 'Dimensions[source_workload_namespace]'; unknown attribute source.workload.namespace
2018-10-01T16:24:45.795884Z	error	failed to evaluate expression for field 'Dimensions[destination_version]'; unknown attribute destination.labels
2018-10-01T16:24:45.795979Z	info	Built new config.Snapshot: id='2'
2018-10-01T16:24:45.796133Z	info	Cleaning up handler table, with config ID:1
gc 3 @2.058s 1%: 0.028+14+11 ms clock, 0.084+0.67/12/21+34 ms cpu, 7->7->4 MB, 8 MB goal, 3 P
2018-10-01T16:24:46.799222Z	info	Publishing 2 events
2018-10-01T16:24:46.800374Z	info	Built new config.Snapshot: id='3'
2018-10-01T16:24:46.801572Z	info	Cleaning up handler table, with config ID:2
2018-10-01T16:24:48.169379Z	info	Publishing 2 events
2018-10-01T16:24:48.171775Z	info	Built new config.Snapshot: id='4'
2018-10-01T16:24:48.171866Z	info	Cleaning up handler table, with config ID:3
2018-10-01T16:24:49.172151Z	info	Publishing 8 events
2018-10-01T16:24:49.174276Z	error	Handler not found: handler='apigee-handler.apigee.istio-system'
2018-10-01T16:24:49.174319Z	error	No valid actions found in rule```

@theganyo can you past full logs here from the beginning?

**Comment by bianpengyuan** [comment]
The log suggests a different problem than what we originally thought. Seems like the handler in rule is not known by mixer. @theganyo Could you make sure the handler resources exist in istio-system namespace? (kubectl get apigee -n istio-system)

**Comment by eyeofthefrog** [comment]
I'm seeing similar issues with 1.0.1 on every restart of the pods....and it's not consistent. I'm using the istio-policy pods for my testing. Sometimes a pod will have no errors in the log, sometimes only 2-3 adapters will have issues with "unknown attributes", and sometimes all the adapters have issues. For example, here are four istio-policy pods running right now:

```
$ kubectl -n istio-system logs istio-policy-78c6f5d49c-5mh4m   mixer |grep error


$ kubectl -n istio-system logs istio-policy-78c6f5d49c-t2gs5  mixer |grep error
2018-10-10T17:24:04.243270Z     error   unknown attribute context.protocol
2018-10-10T17:24:04.243296Z     error   Handler not found: handler='handler.prometheus'
2018-10-10T17:24:04.243301Z     error   No valid actions found in rule
2018-10-10T17:24:04.243320Z     error   unknown attribute context.protocol
2018-10-10T17:24:04.243332Z     error   Instance not found: instance='accesslog.logentry'
2018-10-10T17:24:04.243338Z     error   No valid instances found
2018-10-10T17:24:04.243342Z     error   No valid actions found in rule
2018-10-10T17:24:04.243370Z     error   Handler not found: handler='octarinehandler.octarine'
2018-10-10T17:24:04.243375Z     error   No valid actions found in rule
2018-10-10T17:24:04.243387Z     error   Instance not found: instance='attributes.kubernetes'
2018-10-10T17:24:04.243396Z     error   No valid instances found
2018-10-10T17:24:04.243401Z     error   No valid actions found in rule
2018-10-10T17:24:04.243415Z     error   unknown attribute context.protocol
2018-10-10T17:24:04.243427Z     error   Instance not found: instance='tcpaccesslog.logentry'
2018-10-10T17:24:04.243433Z     error   No valid instances found
2018-10-10T17:24:04.243437Z     error   No valid actions found in rule
2018-10-10T17:24:04.243461Z     error   unknown attribute context.protocol
2018-10-10T17:24:04.243471Z     error   Handler not found: handler='handler.prometheus'
2018-10-10T17:24:04.243482Z     error   No valid actions found in rule
2018-10-10T17:24:04.243506Z     error   unknown attribute context.protocol
2018-10-10T17:24:04.243524Z     error   Instance not found: instance='attributes.kubernetes'
2018-10-10T17:24:04.243531Z     error   No valid instances found
2018-10-10T17:24:04.243542Z     error   No valid actions found in rule
2018-10-10T17:24:05.844599Z     error   failed to evaluate expression for field 'Variables[destinationServiceName]'; unknown attribute destination.service.name
2018-10-10T17:24:05.844636Z     error   failed to evaluate expression for field 'Variables[sourceNamespace]'; unknown attribute source.namespace
2018-10-10T17:24:05.844655Z     error   failed to evaluate expression for field 'Variables[destinationWorkload]'; unknown attribute destination.workload.name
2018-10-10T17:24:05.849781Z     error   unknown attribute context.protocol
2018-10-10T17:24:05.849826Z     error   Instance not found: instance='attributes.kubernetes'
2018-10-10T17:24:05.849840Z     error   No valid instances found
2018-10-10T17:24:05.849851Z     error   No valid actions found in rule
2018-10-10T17:24:05.849867Z     error   Handler not found: handler='octarinehandler.octarine'
2018-10-10T17:24:05.849928Z     error   No valid actions found in rule
2018-10-10T17:24:05.849940Z     error   Instance not found: instance='attributes.kubernetes'
2018-10-10T17:24:05.849949Z     error   No valid instances found
2018-10-10T17:24:05.849954Z     error   No valid actions found in rule
2018-10-10T17:24:05.849980Z     error   unknown attribute context.protocol
2018-10-10T17:24:05.849998Z     error   Handler not found: handler='handler.prometheus'
2018-10-10T17:24:05.850006Z     error   No valid actions found in rule
2018-10-10T16:42:53.620700Z     error   No valid actions found in rule
2018-10-10T17:24:05.850006Z     error   No valid actions found in rule
2018-10-10T17:24:05.850024Z     error   unknown attribute context.protocol
2018-10-10T17:24:05.850040Z     error   Instance not found: instance='accesslog.logentry'
2018-10-10T17:24:05.850047Z     error   No valid instances found
2018-10-10T17:24:05.850052Z     error   No valid actions found in rule
2018-10-10T17:24:05.850068Z     error   unknown attribute context.protocol
2018-10-10T17:24:05.850083Z     error   Instance not found: instance='tcpaccesslog.logentry'
2018-10-10T17:24:05.850094Z     error   No valid instances found
2018-10-10T17:24:05.850098Z     error   No valid actions found in rule
2018-10-10T17:24:05.850114Z     error   unknown attribute context.protocol
2018-10-10T17:24:05.850126Z     error   Handler not found: handler='handler.prometheus'
2018-10-10T17:24:05.850136Z     error   No valid actions found in rule
2018-10-10T17:24:07.043618Z     error   failed to evaluate expression for field 'Variables[destinationWorkloadNamespace]'; unknown attribute destination.workload.namespace
2018-10-10T17:24:07.043650Z     error   failed to evaluate expression for field 'Value'; unknown attribute request.size
2018-10-10T17:24:07.043674Z     error   failed to evaluate expression for field 'Variables[totalReceivedBytes]'; unknown attribute connection.received.bytes_total
2018-10-10T17:24:07.043702Z     error   failed to evaluate expression for field 'Variables[destinationName]'; unknown attribute destination.name
2018-10-10T17:24:07.043731Z     error   failed to evaluate expression for field 'Value'; unknown attribute connection.sent.bytes
2018-10-10T17:24:07.043746Z     error   failed to evaluate expression for field 'Value'; unknown attribute response.size
2018-10-10T17:24:07.043814Z     error   failed to evaluate expression for field 'Value'; unknown attribute connection.received.bytes
2018-10-10T17:24:07.043846Z     error   failed to evaluate expression for field 'Dimensions[destination_version]'; unknown attribute destination.labels
2018-10-10T17:24:07.043863Z     error   failed to evaluate expression for field 'Value'; unknown attribute response.duration
2018-10-10T17:24:07.055897Z     error   unknown attribute context.protocol
2018-10-10T17:24:07.056010Z     error   Handler not found: handler='handler.prometheus'
2018-10-10T17:24:07.056043Z     error   No valid actions found in rule
2018-10-10T17:24:07.056094Z     error   unknown attribute context.protocol
2018-10-10T17:24:07.056123Z     error   Instance not found: instance='tcpaccesslog.logentry'
2018-10-10T17:24:07.056131Z     error   No valid instances found
2018-10-10T17:24:07.056149Z     error   No valid actions found in rule
2018-10-10T17:24:07.056166Z     error   unknown attribute context.protocol
2018-10-10T17:24:07.056192Z     error   Instance not found: instance='attributes.kubernetes'
2018-10-10T17:24:07.056214Z     error   No valid instances found
2018-10-10T17:24:07.056219Z     error   No valid actions found in rule
2018-10-10T17:24:07.056255Z     error   Instance not found: instance='octarinelog.logentry'
2018-10-10T17:24:07.056284Z     error   Instance not found: instance='octarineauth.authorization'
2018-10-10T17:24:07.056301Z     error   No valid instances found
2018-10-10T17:24:07.056308Z     error   No valid actions found in rule
2018-10-10T17:24:07.056325Z     error   Instance not found: instance='attributes.kubernetes'
2018-10-10T17:24:07.056371Z     error   No valid instances found
2018-10-10T17:24:07.056392Z     error   No valid actions found in rule
2018-10-10T17:24:07.056423Z     error   unknown attribute context.protocol
2018-10-10T17:24:07.056434Z     error   Instance not found: instance='accesslog.logentry'
2018-10-10T17:24:07.056450Z     error   No valid instances found
2018-10-10T17:24:07.056455Z     error   No valid actions found in rule
2018-10-10T17:24:07.056490Z     error   unknown attribute context.protocol
2018-10-10T17:24:07.056521Z     error   Handler not found: handler='handler.prometheus'
2018-10-10T17:24:07.056541Z     error   No valid actions found in rule


$ kubectl -n istio-system logs istio-policy-78c6f5d49c-xr52b  mixer |grep error
2018-10-10T17:26:27.072553Z     error   Handler not found: handler='handler.stdio'
2018-10-10T17:26:27.072577Z     error   No valid actions found in rule
2018-10-10T17:26:27.072593Z     error   Handler not found: handler='handler.kubernetesenv'
2018-10-10T17:26:27.072604Z     error   No valid actions found in rule
2018-10-10T17:26:27.072624Z     error   Instance not found: instance='octarinelog.logentry'
2018-10-10T17:26:27.072635Z     error   Instance not found: instance='octarineauth.authorization'
2018-10-10T17:26:27.072658Z     error   No valid instances found
2018-10-10T17:26:27.072664Z     error   No valid actions found in rule
2018-10-10T17:26:27.072688Z     error   Handler not found: handler='handler.stdio'
2018-10-10T17:26:27.072701Z     error   No valid actions found in rule
2018-10-10T17:26:27.072730Z     error   Handler not found: handler='handler.prometheus'
2018-10-10T17:26:27.072763Z     error   No valid actions found in rule
2018-10-10T17:26:27.072788Z     error   Handler not found: handler='handler.kubernetesenv'
2018-10-10T17:26:27.072803Z     error   No valid actions found in rule
2018-10-10T17:26:27.072825Z     error   Handler not found: handler='handler.prometheus'
2018-10-10T17:26:27.072846Z     error   No valid actions found in rule
2018-10-10T17:26:28.466535Z     error   Handler not found: handler='handler.stdio'
2018-10-10T17:26:28.466564Z     error   No valid actions found in rule
2018-10-10T17:26:28.466595Z     error   Instance not found: instance='octarineauth.authorization'
2018-10-10T17:26:28.466616Z     error   Handler not found: handler='handler.stdio'
2018-10-10T17:26:28.466625Z     error   No valid actions found in rule
2018-10-10T17:26:29.613465Z     error   Handler not found: handler='handler.stdio'
2018-10-10T17:26:29.613557Z     error   No valid actions found in rule
2018-10-10T17:26:29.613664Z     error   Handler not found: handler='handler.stdio'
2018-10-10T17:26:29.613685Z     error   No valid actions found in rule
2018-10-10T17:26:29.718423Z     error   adapters        adapter did not close all the scheduled daemons {"adapter": "handler.kubernetesenv.istio-system"}
2018-10-10T17:26:29.718440Z     error   adapters        adapter did not close all the scheduled daemons {"adapter": "handler.prometheus.istio-system"}
2018-10-10T17:26:32.176601Z     error   adapters        adapter did not close all the scheduled daemons {"adapter": "handler.kubernetesenv.istio-system"}


$ kubectl -n istio-system logs istio-policy-78c6f5d49c-ddnh4  mixer |grep error
2018-10-10T17:30:46.841025Z     error   failed to evaluate expression for field 'Variables[sourceVersion]'; unknown attribute source.labels
2018-10-10T17:30:46.841094Z     error   failed to evaluate expression for field 'Subject'; failed to evaluate expression for field 'Subject.User': unknown attribute request.auth.principal
2018-10-10T17:30:46.841117Z     error   failed to evaluate expression for field 'Variables[sourceIp]'; unknown attribute source.ip
2018-10-10T17:30:46.841132Z     error   failed to evaluate expression for field 'Variables[clientTraceId]'; unknown attribute request.headers
2018-10-10T17:30:50.827461Z     error   Instance not found: instance='requestcount.metric'
2018-10-10T17:30:50.827507Z     error   Instance not found: instance='requestduration.metric'
2018-10-10T17:30:50.827531Z     error   Instance not found: instance='requestsize.metric'
2018-10-10T17:30:50.827544Z     error   Instance not found: instance='responsesize.metric'
2018-10-10T17:30:50.827552Z     error   No valid instances found
2018-10-10T17:30:50.827558Z     error   No valid actions found in rule
2018-10-10T17:30:50.827600Z     error   Instance not found: instance='tcpbytesent.metric'
2018-10-10T17:30:50.827618Z     error   Instance not found: instance='tcpbytereceived.metric'
2018-10-10T17:30:50.827626Z     error   No valid instances found
2018-10-10T17:30:50.827647Z     error   No valid actions found in rule
2018-10-10T17:30:52.134506Z     error   adapters        adapter did not close all the scheduled daemons {"adapter": "handler.kubernetesenv.istio-system"}
```

**Comment by bianpengyuan** [comment]
@eyeofthefrog Do those errors persist and make mixer stop working, or they just showed up in initialization and was eventually resolved? If it is the latter then the errors are expected. Mixer used to wait 30s at initialization for all CRs. Now it does not wait and processes CRs whenever they reaches mixer, which means there could be race like instance CRs come into mixer before attributes manifest.

**Comment by eyeofthefrog** [comment]
@bianpengyuan It never works if I see the initial error. 

When you say the mixer used to "wait" for the CRDs, what do you mean exactly? Does it have to wait on the already loaded CRDs to be read from the system.....or is it waiting on newly loaded CRDs to be available? My Istio cluster is a week old. Just restarting the policy or telemetry pods still results in the "unknown attribute" issue without adding or removing any CRDs.

**Comment by eyeofthefrog** [comment]
I reinstalled everything and I'm unable to reproduce this issue. I'll play around with it more, but I don't think there's anything to look into deeper for now. Thanks.

**Comment by cmluciano** [comment]
Is this issue still relevant?
